File: ios/ExpoManagedNativeCode.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ExpoManagedNativeCode, NSObject)

RCT_EXTERN_METHOD(sayHello:(NSString)a withB:(NSString)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/ExpoManagedNativeCode.swift
@objc(ExpoManagedNativeCode)
class ExpoManagedNativeCode: NSObject {

  @objc(multiply:withB:withResolver:withRejecter:)
  func sayHello(a: String, b: String, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
     resolve("Hello \(a), and hello \(b) from Swift!")
  }
}
